### PR TITLE
Test:  fix ZIP entry LastWriteTime tests to support running locally (when system clock is not UTC)

### LINKED ISSUE #11312: Test_ExtractionHonorsFileTimestamp() and ExtractPackageAsync_PreservesZipEntryTimeAsync() should work on any timezone
Test_ExtractionHonorsFileTimestamp() and ExtractPackageAsync_PreservesZipEntryTimeAsync() currently only work for the timezone used by our CI. These tests should be able to run on any timezone so that they can pass locally for any contributor.

### PR DESCRIPTION
**Pull request by dtivel**
# Bug

Fixes: https://github.com/NuGet/Home/issues/11312

## Description

This change fixes some tests that only pass when the system clock is UTC (like on a CI build machine).

The way the `DateTimeOffset ZipArchiveEntry.LastWriteTime` property works is for any `DateTimeOffset` value you set it to, `ZipArchiveEntry` will use only the `DateTime` property of that `DateTimeOffset`.  [`DateTimeOffset.DateTime`](https://learn.microsoft.com/dotnet/api/system.datetimeoffset.datetime?view=net-8.0) effectively returns the `DateTime` component of a `DateTimeOffset`, after first discarding its time zone information.  To illustrate, the `DateTime` property of these different `DateTimeOffset` values all return the same `DateTime` value of `8/3/2024 11:50:51 AM` with an "unspecified" time zone.

```
8/3/2024 11:50:51 AM -07:00
8/3/2024 11:50:51 AM +00:00
8/3/2024 11:50:51 AM +05:30
```

That means any test that checks that NuGet sets a file's `LastWriteTime` value correctly must use `DateTimeOffset.DateTime` to check the file's last write time in UTC, or `DateTimeOffset.DateTime.ToLocalTime()` to check the file's last write time in local.

As part of this change, I changed the `entryModifiedTime` type from `DateTime` to `DateTimeOffset` to match the `ZipArchiveEntry.LastWriteTime` data type.

I also opportunistically fixed some code style issues in the area.

## PR Checklist

- [X] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] ~~Added tests~~ N/A
- [ ] ~~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~~ N/A

CC @aortiz-msft

### CODE REVIEW

File: test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageArchiveReaderTests.cs [hunk 1]
```diff
@@ -5,7 +5,9 @@
 using System.IO;
 using System.IO.Compression;
 using System.Linq;
+#if IS_SIGNING_SUPPORTED
```
Comment (dtivel) [inline test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageArchiveReaderTests.cs:8]: These `#if`'s resolve IDE warnings that some `using` statements are unnecessary under some TFM's.

File: test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs [hunk 2]
```diff
@@ -27,19 +30,19 @@
 
 namespace NuGet.Packaging.Test
 {
-    using LocalPackageArchiveDownloader = NuGet.Protocol.LocalPackageArchiveDownloader;
+    using LocalPackageArchiveDownloader = Protocol.LocalPackageArchiveDownloader;
 
     [Collection(SigningTestsCollection.Name)]
     public class PackageExtractorTests
     {
-        private static ClientPolicyContext _defaultContext = ClientPolicyContext.GetClientPolicy(NullSettings.Instance, NullLogger.Instance);
```
Comment (dtivel) [inline test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs:35]: These renames resolve style warnings that `static` or `const` fields shouldn't be `_`-prefixed.

File: test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs [hunk 3]
```diff
@@ -1212,21 +1215,24 @@ public async Task ExtractPackageAsync_WithoutPackageSaveModeNupkg_DoesNotExtract
...
+            using (TestDirectory root = TestDirectory.Create())
             {
-                var time = DateTime.Parse("2014-09-26T01:23:00Z",
-                    System.Globalization.CultureInfo.InvariantCulture,
-                    System.Globalization.DateTimeStyles.AdjustToUniversal);
+                DateTimeOffset time = DateTimeOffset.Parse("2014-09-26T01:23:00Z",
+                        CultureInfo.InvariantCulture,
+                        DateTimeStyles.AdjustToUniversal)
+                    .ToLocalTime();
+                DateTime expectedLastWriteTimeUtc = time.DateTime;
```
Comment (dtivel) [inline test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtractorTests.cs:1224]: This is the actual test fix.

File: test/NuGet.Core.Tests/NuGet.Packaging.Test/NugetPackageUtilTests.cs [hunk 4]
```diff
@@ -620,7 +620,9 @@ public async Task Test_ExtractionHonorsFileTimestamp()
         {
             // Arrange
             var packageIdentity = new PackageIdentity("packageA", new NuGetVersion("2.0.3"));
-            var entryModifiedTime = new DateTimeOffset(1985, 11, 20, 12, 0, 0, TimeSpan.FromHours(-7.0)).DateTime;
+            var entryModifiedTime = new DateTimeOffset(1985, 11, 20, 12, 0, 0, TimeSpan.FromHours(-7.0));
+            DateTime expectedLastWriteTime = entryModifiedTime.DateTime.ToLocalTime();
```
Comment (dtivel) [inline test/NuGet.Core.Tests/NuGet.Packaging.Test/NugetPackageUtilTests.cs:624]: This is the actual test fix.

**Review by jeffkl — APPROVED** [review]
Thanks for fixing this, I saw fail a few months ago and never got around to doing it.

**Review by zivkan — APPROVED** [review]

**Pull request merged**